app/assets/javascripts/application.js.erb

```diff
@@ -515,16 +465,41 @@ const onPageLoad = (callback) => {
 const replace = (oldElement, newHTMLString) => {
   const placeHolder = document.createElement('div');
   placeHolder.insertAdjacentHTML('afterBegin', newHTMLString);
-  const newElement = placeHolder.firstElementChild;
-  oldElement.replaceWith(newElement);
+  const newElements = placeHolder.childNodes.values();
+  oldElement.replaceWith(...newElements);
+}
+
+const fetchWithCSRF = (url, params) => {
+  let csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+  let h = new Headers();
+  h.append('X-CSRF-Token', csrfToken);
+  params = params || {};
+  params['headers'] = h;
```

Comment:
```suggestion
  params = params || {};
  params['headers'] = params['headers'] || new Headers();
  params['headers'].append('X-CSRF-Token', csrfToken);
```
This overwrites all headers set by the caller, so I reused your `||` trick from the line above and rearranged to have the function only replace the CSRF header. It's not a bug yet, or even likely to be a bug, but it would be a pretty frustrating one to run down if it happened.

Reply:
Thank you for catching the oversight.